### DISCUSSION TITLE
Being able to export NextAuthRequest from next-auth

### DISCUSSION THREAD

**Discussion by pearson974** [nextauthjs/next-auth#11365] [Ideas]
### Goals

1. Currently, when trying to wrap an API route with auth, it's complaining that the request type doesn't have auth type because NextAuthRequest cannot be imported
2.
3.

### Non-Goals

1.
2.
3.

### Background

The scenario where we need to access the session inside of an API route handler in Next.js is a common scenario. The recommended way of doing it is by using the auth function to wrap and API route handler. The request parameter will then have an auth key on it which you can check for a valid session. However, TypeScript is complaining about the auth not existing on request because we currently cannot import ```NextAuthRequest```.

### Proposal

Export ```NextAuthRequest``` to ```next-auth/src/index.ts```.

**Comment by lorenzo-dallamuta** [comment]
For the time being we could just copypaste from the NextAuth source code into ours:

```
import { Session } from "next-auth";

// node_modules/next-auth/src/lib/index.ts
interface NextAuthRequest extends NextRequest {
  auth: Session | null;
}
```

Is there a motivation for the typing of NextAuthRequest to not be accessible by the users of the library? Is it intended to discourage abstractions in Next.js middleware handler?
